draft-ietf-ppm-dap.md

```diff
@@ -2686,50 +2575,115 @@ collude and break robustness. Is this a contingency we need to address? There
+Aggregators refuse shared parameters that are trivially insecure (e.g., a
+minimum batch size of 1 report).
 
-An important parameter of a DAP deployment is the minimum batch size. If an
-aggregation includes too few inputs, then the outputs can reveal information
-about individual participants. Aggregators use the batch size field of the
-shared task parameters to enforce minimum batch size during the collect
-protocol, but server implementations may also opt out of participating in a DAP
-task if the minimum batch size is too small. This document does not specify how
-to choose minimum batch sizes.
+### Verification key requirements {#verification-key}
 
-The DAP parameters also specify the maximum number of times a report can be
-used. Some protocols, such as Poplar {{BBCGGI21}}, require reports to be used in
-multiple batches spanning multiple collect requests.
+The verification key for a task SHOULD be chosen before any reports are
+generated. It SHOULD be fixed for the lifetime of the task and not be rotated.
+One way to ensure this is to include the verification key in a derivation of the
+task ID.
+
+This consideration comes from current security analysis for existing VDAFs. For
+example, to ensure that the security proofs for Prio3 hold, the verification key
+MUST be chosen independently of the generated reports. This can be achieved as
+recommended above.
+
+### Batch parameters
+
+An important parameter of a DAP deployment is the minimum batch size. If a batch
+includes too few reports, then the aggregate result can reveal information
+about individual participants. Aggregators must enforce the agreed-upon minimum
+batch size during the collect protocol, but implementations may also opt out of
+participating in a DAP task if the minimum batch size is too small. This
+document does not specify how to choose minimum batch sizes.
+
+### VDAFs and compute requirements
+
+The choice of VDAF can impact the computation required for a DAP Task. For
+instance, the Poplar1 VDAF {{!VDAF}} when configured to compute a set of heavy
+hitters requires each measurement to be of the same bit-length which all parties
+need to agree on prior to VDAF execution. The computation required for such
+tasks can increase superlinearly as multiple rounds of evaluation are needed for
+each bit of the measurement value.
+
+When dealing with variable length measurements (e.g domain names), it is
+necessary to pad them to convert into fixed-size measurements. When computing
+the heavy hitters from a batch of such measurements, we can early-abort the
+Poplar1 execution once we have reached the padding region for a candidate
+measurement. For smaller length measurements, this significantly reduces the
+cost of communication between Aggregators and the steps required for the
+computation. However, malicious Clients can still generate maximum length
+measurements forcing the system to always operate at worst-case performance.
```

Comment:
This is an existing paragraph, moved around by this PR, but could you add an open issue note pointing to cfrg/draft-irtf-cfrg-vdaf#273? In short, optimized Poplar1 tree walking algorithms should skip to the leaf level and run one last aggregation, rather than stopping after any inner level, in order to maintain robustness.